src/coreclr/gc/gc.cpp

```diff
@@ -11712,7 +11712,7 @@ void gc_heap::init_heap_segment (heap_segment* seg, gc_heap* hp
 #endif //USE_REGIONS
     )
 {
-    seg->flags = 0;
+    seg->flags = seg->flags & heap_segment_flags_ma_committed;
```

Comment:
you can't just turn on the flag on, it can only be turned on if the mark array is actually committed. it should be preserved, meaning if it's turned on it should not be be turned off unless we decommit the mark array (which we don't do (yet)). it would also be good to do a debug verification that if this flag is on, all mark array bits in this region are cleared.

Reply:
my bad - totally misread that as '|'